lib/accessly/permitted_actions/on_object_query.rb

```diff
@@ -31,6 +31,38 @@ def can?(action_id, object_type, object_id)
             ).exists?
         end
       end
+
+      # Returns an ActiveRecord::Relation of object_type containing the
+      # records on which the actor has permission to perform action_id.
+      #
+      # @param action_id [Integer] The action we're checking on the actor for the object.
+      # @param object_type [ActiveRecord::Base] The ActiveRecord model to be loaded.
```

Comment:
If we do opt for the duck type idea, we could make the type here `[::where]` to signify that. 

I'm wondering now if we should call this something like "scope" or "object" or "class". Maybe scope fits it best... it's basically just a thing that we can ask "where" with the given ids which could be a class... could be a instance of active record relation for that matter like `Lesson.unarchived`. In that case... I guess the type would accept anything that response to `[::where, #where]`? What's your thoughts around that?